In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


# Load the Data

In [9]:
observations = pd.read_csv('https://raw.githubusercontent.com/cdejong1/National_Park_Biodiversity/main/biodiversity_starter/observations.csv',encoding='utf-8')
species = pd.read_csv('https://raw.githubusercontent.com/cdejong1/National_Park_Biodiversity/main/biodiversity_starter/species_info.csv',encoding='utf-8')
print(observations.head())
print(species.head())
print(species.dtypes)

            scientific_name                            park_name  observations
0        Vicia benghalensis  Great Smoky Mountains National Park            68
1            Neovison vison  Great Smoky Mountains National Park            77
2         Prunus subcordata               Yosemite National Park           138
3      Abutilon theophrasti                  Bryce National Park            84
4  Githopsis specularioides  Great Smoky Mountains National Park            85
  category                scientific_name  \
0   Mammal  Clethrionomys gapperi gapperi   
1   Mammal                      Bos bison   
2   Mammal                     Bos taurus   
3   Mammal                     Ovis aries   
4   Mammal                 Cervus elaphus   

                                        common_names conservation_status  
0                           Gapper's Red-Backed Vole                 NaN  
1                              American Bison, Bison                 NaN  
2  Aurochs, Aurochs, Domestic 

# Explore the Data

In [12]:
#find the number of unique species 
print(f"number of species:{species.scientific_name.nunique()}")

number of species:5541


In [13]:
# find the number of categories
print(f"number of categories:{species.category.nunique()}")
print(f"categories:{species.category.unique()}")

number of categories:7
categories:['Mammal' 'Bird' 'Reptile' 'Amphibian' 'Fish' 'Vascular Plant'
 'Nonvascular Plant']


In [15]:
species.groupby('category').size()

category
Amphibian              80
Bird                  521
Fish                  127
Mammal                214
Nonvascular Plant     333
Reptile                79
Vascular Plant       4470
dtype: int64

In [23]:
#find the number of unique conservation_status
print(f"number of conservation status's: {species.conservation_status.nunique()}")
print(f"conservation status's possible: {species.conservation_status.unique()}")
print(species.groupby('conservation_status').size())
print(f"na values: {species.conservation_status.isna().sum()}")

number of conservation status's: 4
conservation status's possible: [nan 'Species of Concern' 'Endangered' 'Threatened' 'In Recovery']
conservation_status
Endangered             16
In Recovery             4
Species of Concern    161
Threatened             10
dtype: int64
na values: 5633


There are 5,633 nan values which means that they are species without concerns. On the other hand there are 161 species of concern, 16 endangered, 10 threatened, and 4 in recovery.

Now we do the same for the observation data set 

In [26]:
#find the number of unique national parks
print(f"number of parks: {observations.park_name.nunique()}")
print(f"park names: {observations.park_name.unique()}")

number of parks: 4
park names: ['Great Smoky Mountains National Park' 'Yosemite National Park'
 'Bryce National Park' 'Yellowstone National Park']


In [27]:
#find the total number of observations
print(f"number of observations: {observations.observations.sum()}")

number of observations: 3314739


# Analysis